byterestclient/__init__.py

```diff
@@ -35,12 +35,19 @@ def request(self, method, path, data=None, *args, **kwargs):
             **kwargs
         )
 
-        response.raise_for_status(allow_redirects=False)
+        response.raise_for_status()
+
+        if 300 <= response.status_code < 400:
+            # support 0.12.1 by setting response seperately
+            exception = HTTPError('%s Encountered Redirect' % response.status_code)
+            exception.response = response
+            raise exception
```

Comment:
What about `304 Not Modified`? The exception description is misleading because 304 is not a redirect. I'm not sure triggering an exception on 304 is the best way to go here (but can't immediately offer a better approach).